test/e2e_node/util.go

```diff
@@ -170,6 +170,21 @@ func getCurrentKubeletConfig(ctx context.Context) (*kubeletconfig.KubeletConfigu
 	return e2enodekubelet.GetCurrentKubeletConfig(ctx, framework.TestContext.NodeName, "", false, framework.TestContext.StandaloneMode)
 }
 
+func cleanupPods(f *framework.Framework) {
+	ginkgo.AfterEach(func(ctx context.Context) {
+		ginkgo.By("Deleting any Pods created by the test in namespace: " + f.Namespace.Name)
+		l, err := e2epod.NewPodClient(f).List(ctx, metav1.ListOptions{})
+		framework.ExpectNoError(err)
+		for _, p := range l.Items {
+			if p.Namespace != f.Namespace.Name {
+				continue
+			}
+			framework.Logf("Deleting pod: %s", p.Name)
+			e2epod.NewPodClient(f).DeleteSync(ctx, p.Name, metav1.DeleteOptions{}, 2*time.Minute)
```

Comment:
do you want to keep it sync? This can be optimized to issue all delete commands first and then check for a status

Reply:
I see this pattern extensively used and since this is a Serial test and running in a CI job with a single ginkgo node and we have a big enough timeout (and we know it will take time..) for all the reasons, will leave this as is for now:
https://cs.k8s.io/?q=e2epod.NewPodClient.*.DeleteSync%5C(&i=nope&files=&excludeFiles=&repos=

Will watch a few runs and we can optimize if needed.